src/Microsoft.SqlTools.ServiceLayer/Scripting/ScripterCore.cs

```diff
@@ -268,31 +263,44 @@ internal DefinitionResult GetDefinitionUsingDeclarationType(DeclarationType type
         /// <param name="objectType">Type of SQL object</param>
         /// <returns>Location object representing URI and range of the script file</returns>
         internal Location[] GetSqlObjectDefinition(
-                ScriptGetter sqlScriptGetter,
                 string objectName,
                 string schemaName,
                 string objectType)
         {
-            StringCollection scripts = sqlScriptGetter(objectName, schemaName, null);
+            // script file destination
             string tempFileName = (schemaName != null) ? Path.Combine(this.tempPath, string.Format("{0}.{1}.sql", schemaName, objectName))
                                                 : Path.Combine(this.tempPath, string.Format("{0}.sql", objectName));
 
-            if (scripts != null)
+            ScriptingScriptOperation operation = InitScriptOperation(objectName, schemaName, objectType, tempFileName);
+            operation.Execute();
+
+            // If the scripting was successful, we should see a file with the scripts
+            bool objectFound = false;
+            int lineNumber = 0;
```

Comment:
>lineNumber [](start = 16, length = 10)

why lineNumber and lineCount variables in same function?  I guess lineNumber is something like lineNumberOfCreateStatement or something like that.  Please use a better name that is ambiguous with other variables in function